src/types/index.ts

```diff
@@ -0,0 +1,184 @@
+import { MyInfoPassType } from "./passtype"
+import { MyInfoPassStatus } from "./passstatus"
+import { MyInfoPassExpiryDate } from "./passexpirydate"
+import { MyInfoEmploymentSector } from "./employmentsector"
+
+type IPersonFull = {
+  uinfin: MyInfoUinFin
+  name: MyInfoName
+  hanyupinyinname: MyInfoHanyuPinyinName
+  aliasname: MyInfoAliasName
+  hanyupinyinaliasname: MyInfoHanyuPinyinAliasName
+  marriedname: MyInfoMarriedName
+  sex: MyInfoSex
+  race: MyInfoRace
+  secondaryrace: MyInfoSecondaryRace
+  dialect: MyInfoDialect
+  nationality: MyInfoNationality
+  dob: MyInfoDob
+  birthcountry: MyInfoBirthCountry
+  residentialstatus: MyInfoResidentialStatus
+  passportnumber: MyInfoPassportNumber
+  passportexpirydate: MyInfoPassportExpiryDate
+  regadd: MyInfoAddress
+  housingtype: MyInfoHousingType
+  hdbtype: MyInfoHdbType
+  hdbownership: MyInfoHdbOwnership[]
+  ownerprivate: MyInfoOwnerPrivate
+  email: MyInfoEmail
+  mobileno: MyInfoPhoneNumber
+  marital: MyInfoMarital
+  marriagecertno: MyInfoMarriageCertNo
+  countryofmarriage: MyInfoCountryOfMarriage
+  marriagedate: MyInfoMarriageDate
+  divorcedate: MyInfoDivorceDate
+  childrenbirthrecords: MyInfoChildrenBirthRecords
+  sponsoredchildrenrecords: MyInfoSponsoredChildrenRecords
+  occupation: MyInfoOccupation
+  employment: MyInfoEmployment
+  passtype: MyInfoPassType
+  passstatus: MyInfoPassStatus
+  passexpirydate: MyInfoPassExpiryDate
+  employmentsector: MyInfoEmploymentSector
+  householdincome: MyInfoHouseholdIncome
+  vehicles: MyInfoVehicle[]
+  drivinglicence: MyInfoDrivingLicence
+  merdekagen: MyInfoMerdekaGen
+  silversupport: MyInfoSilverSupport
+  gstvoucher: MyInfoGstVoucher
+  'noa-basic': MyInfoNoaBasic
+  noa: MyInfoNoa
+  'noahistory-basic': MyInfoNoaHistoryBasic
+  noahistory: MyInfoNoaHistory
+  cpfcontributions: MyInfoCpfContributions
+  cpfemployers: MyInfoCpfEmployers
+  cpfbalances: MyInfoCpfBalances
+}
+
+/**
+ * Shape of data returned by the Person API.
+ */
+export type IPerson = Partial<IPersonFull>
+
+/**
+ * Valid scopes (requested attributes) to get from MyInfo.
+ */
+export type MyInfoScope =
+  | Exclude<
+      keyof IPerson,
+      | MyInfoAttribute.HDBOwnership
+      | MyInfoAttribute.ChildrenBirthRecords
+      | MyInfoAttribute.SponsoredChildrenRecords
+      | MyInfoAttribute.Vehicles
+      | MyInfoAttribute.DrivingLicence
+    >
+  | HdbOwnershipScope
+  | ChildrenBirthRecordsScope
+  | SponsoredChildrenRecordsScope
+  | VehiclesScope
+  | DrivingLicenceScope
+
+// Check that IPerson includes all keys from MyInfoAttribute
+type IPersonCheck = Exclude<MyInfoAttribute, keyof IPerson>
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+type VerifyIPersonCheck<
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  Missing extends never = IPersonCheck
+> = never
+
+export * from './address'
+export * from './aliasname'
+export * from './base'
+export * from './birthcountry'
+export * from './childrenbirthrecords'
```

Comment:
I'm wondering if we're exporting so much from this library that it might get confusing for users. for example, `childrenbirthrecords` exports both `ChildCustomFields` and `MyInfoChildrenBirthRecords`. will the user ever have to use `ChildCustomFields` by itself, and if not, might it be better to just export `MyInfoChildrenBirthRecords`?

Reply:
good point, let me crawl through again and see which ones have more value to be exported.